src/kafka.rs

```diff
@@ -274,8 +275,24 @@ pub async fn setup_integration() {
     let mut stream = consumer.stream();
 
     while let Ok(curr) = stream.next().await.unwrap() {
-        if let Err(err) = ingest_message(curr).await {
-            error!("Unable to ingest incoming kafka message- {err}")
-        }
+        // Constructs a log for each kafka request
+        let mut log_builder = AuditLogBuilder::default();
+        log_builder.set_actor(
+            CONFIG.parseable.kafka_host.as_deref().unwrap_or(""),
+            "Kafka Client",
+            "",
+            "",
```

Comment:
The builder pattern could be more elegant here.
 I would:

-  Chain the builder calls like `.set_actor().set_request().set_stream()` and be sure that prefixes like "set", "with", etc compatible with other builders in the codebase.

-  Handle optional or default values inside builder methods
-  Add precondition checks in build()/send() if validation logics  requires

```rust
AuditLogBuilder::default()
    .set_actor(kafka_host.unwrap_or_default(), "Kafka Client")
    .set_request_type("Kafka")
    .set_stream(curr.topic())
    .set_response(status, err.as_str())
    .send()
    .await